foundation/foundation-upgrade/src/main/java/com/boozallen/aissemble/upgrade/migration/v1_10_0/PomDependencyVersionMigration.java

```diff
@@ -0,0 +1,147 @@
+            "junit:junit:${version.junit}",
+            "org.jboss.resteasy:resteasy-client:${version.resteasy}",
+            "org.jboss.resteasy:resteasy-jackson2-provider:${version.resteasy}",
+            "org.awaitility:awaitility:${version.awaitility}",
+            "io.vertx:vertx-core:${version.vertx}",
+            "io.smallrye.config:smallrye-config:${version.smallrye.config}",
+            "org.slf4j:slf4j-api:${version.slf4j}",
+            "org.slf4j:slf4j-simple:${version.slf4j}"
+        };
+
+        return Arrays.stream(dependencyStrings)
+            .map(dependencyString -> {
+                String[] dependencyAttributes = dependencyString.split(":");
+                Dependency dependency = new Dependency();
+                dependency.setGroupId(dependencyAttributes[0]);
+                dependency.setArtifactId(dependencyAttributes[1]);
+                dependency.setVersion(dependencyAttributes[2]);
+                return dependency;
+            })
+            .toList();
+    }
+
+    /**
+     * Returns a list of any dependencies that need to be updated with a new version
+     */
+    private List<Dependency> getMatchingDependenciesWithDifferentVersionsForProject(Model model) {
+        List<Dependency> matchingDependencies = new ArrayList<>();
+
+        this.dependenciesToUpdate.stream()
+            .flatMap(updatedDependency -> 
+                this.getAllDependenciesByFilterForProject(model, dependency -> this.isMatchingDependencyWithDifferentVersion(dependency, updatedDependency))
+                .stream())
+            .forEach(matchingDependencies::add);
+
+        return matchingDependencies;
+    }
+
+    /**
+     * Checks if given dependency matches group id and artifact id but not version of the updated dependency
+     */
+    private boolean isMatchingDependencyWithDifferentVersion(Dependency dependency, Dependency updatedDependency) {
+        return this.isMatchingDependency(dependency, updatedDependency) &&
+               // Return true if the version is not specified (null) or the version is specified but different
+               (dependency.getVersion() == null || !dependency.getVersion().equals(updatedDependency.getVersion()));
+    }
+
+    /**
+     * Checks if two given dependencies match group id and artifact id
+     */
+    private boolean isMatchingDependency(Dependency dependency1, Dependency dependency2) {
+        return dependency1.getGroupId().equals(dependency2.getGroupId()) &&
+               dependency1.getArtifactId().contains(dependency2.getArtifactId());
+    }
+
+    /**
+     * Returns the updated version for the matching dependency
+     */
+    private String getUpdatedDependencyVersion(Dependency dependency) {
+        return this.dependenciesToUpdate.stream()
+            .filter(updatedDependency -> this.isMatchingDependency(updatedDependency, dependency))
+            .map(Dependency::getVersion)
+            .findFirst()
+            .orElse(null);
```

Comment:
S: Looks like the calling code assumes non-null. Should we use `orElseThrow` instead?

Reply:
Yes that should throw an error if we don't find the version.